consensus/src/round_manager.rs

```diff
@@ -913,4 +930,25 @@ impl RoundManager {
             .round(self.round_state.current_round())
             .epoch(self.epoch_state.epoch)
     }
+
+    /// Given R1 <- B2 if R1 has the reconfiguration txn, we inject error on B2 if R1.round + 1 = B2.round
+    /// Direct suffix is checked by parent.has_reconfiguration && !parent.parent.has_reconfiguration
+    ///
+    /// It's only enabled with fault injection (failpoints feature).
+    #[allow(unused_variables)]
+    fn should_inject_reconfiguration_error(block_data: &BlockData) -> bool {
+        #[cfg(feature = "failpoints")]
+        {
+            let direct_suffix = block_data.is_reconfiguration_suffix()
+                && !block_data
+                    .quorum_cert()
+                    .parent_block()
+                    .has_reconfiguration();
+            let continuous_round =
+                block_data.round() == block_data.quorum_cert().certified_block().round() + 1;
+            direct_suffix && continuous_round
```

Comment:
How about give this 50% chance of failing?

Reply:
are you worried that happy path is not tested? we have plenty of chances testing that e.g. cluster test runs a 100 epoch tests without failpoints.